drawBot/drawBotDrawingTools.py

```diff
@@ -1978,6 +1978,33 @@ def imagePixelColor(self, path, xy):
         color = color.colorUsingColorSpaceName_("NSCalibratedRGBColorSpace")
         return color.redComponent(), color.greenComponent(), color.blueComponent(), color.alphaComponent()
 
+    def imageResolution(self, path):
+        """
+        Return the image resolution for a given image.
+        """
+        if isinstance(path, AppKit.NSImage):
+            # its an NSImage
+            # get the bitmap representation
+            try:
+                rep = path.representations()[0]
+            except Exception:
+                raise DrawBotError("Cannot extract bitmap data from given nsImage object")
+        else:
+            if isinstance(path, basestring):
+                path = optimizePath(path)
+            if path.startswith("http"):
+                url = AppKit.NSURL.URLWithString_(path)
+            else:
+                if not os.path.exists(path):
+                    raise DrawBotError("Image does not exist")
+                url = AppKit.NSURL.fileURLWithPath_(path)
+                try:
+                    rep = AppKit.NSImageRep.imageRepWithContentsOfURL_(url)
+                except Exception:
```

Comment:
Can this exception be more specific, too?

Reply:
this catches a range of errors:  file not found, file not readable, file not valid, ...